### ISSUE TITLE
[Feature Request] Add offsets to token classification prediction tasks

### ISSUE THREAD

**Issue by DrissiReda** [kserve/kserve#4528]
/kind feature

**Describe the solution you'd like**
For token_classification tasks, only the label ids and potentially the probabilities are returned, 
however another very important piece of information isn't which is the offsets that give the start and end position of the detected label. The user will then have to manually extract them by loading the tokenizer locally
and matching them to the ids like so:

```python
tokenizer = AutoTokenizer.from_pretrained(MODEL_REPO, trust_remote_code=True)
enc = tokenizer(text, return_offsets_mapping=True, truncation=True, return_attention_mask=False)
tokens = tokenizer.convert_ids_to_tokens(enc["input_ids"])
# this is where start/end positions are
offsets = enc["offset_mapping"]
```

Also is it possible to choose to return either label names or their ids?
We would probably have some issues with the tokenizers mismatch but this doesn't
have to be the default behavior, and when activated it would take the default tokenizer from the model
like in my example.

Is there any reason this hasn't been done?
